### ISSUE TITLE
Operator with TLS doesn't seem to be working

### ISSUE THREAD

**Issue by bigbosst** [rabbitmq/messaging-topology-operator#469]
## Describe the bug
This is related to #233 - Cannot use operator with tls enabled rabbitmq cluster

{"level":"error","ts":1665071770.8476884,"msg":"Reconciler error","controller":"permission","controllerGroup":"rabbitmq.com","controllerKind":"Permission","permission":{"name":"data-consumer-rabbit-user-permission","namespace":"rabbitmq"},"namespace":"rabbitmq","name":"data-consumer-rabbit-user-permission","reconcileID":"xxxxxxxxx","error":"Put \"https://rabbitmq-cluster.rabbitmq.svc:15671/api/permissions/%2F/data-consumer-rabbit-user\": x509: certificate is valid for rabbitmq-cluster.my.domain.com, not rabbitmq-cluster.rabbitmq.svc","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.3/pkg/internal/controller/controller.go:234"}

## To Reproduce
I patched the MTO following these instructions with no issues:
https://www.rabbitmq.com/kubernetes/operator/tls-topology-operator.html

I attempted to apply a patch for MESSAGING_DOMAIN_NAME documented here:
https://www.rabbitmq.com/kubernetes/operator/install-topology-operator.html

I got this error while trying to patch the MESSAGING_DOMAIN_NAME :
```
$ kubectl -n rabbitmq-system patch deployment messaging-topology-operator --patch "spec:
  template:
    spec:
      containers:
      - command:
        - /manager
        env:
        - name: MESSAGING_DOMAIN_NAME
          value: "my.domain.com" 
"

```
Error from server: map: map[command:[/manager] env:[map[name:MESSAGING_DOMAIN_NAME value:my.domain.com]]] does not contain declared merge key: name

Unable to apply the patch, I instead edited the YAML directly using kubectl edit. I was able to add the additional env variable this way, however even after redeploying the messaging topology operator pod, it doesn't seem to make a difference. It still gives the same error message failing the cert because it's still using the svc hostname.

I only noticed the SSL error messages when trying to add a new user/permission failed. 

**Expected behavior**
Should be using a hostname ending with '.my.domain.com' 

## Version and environment information
- Messaging Topology Operator: rabbitmqoperator/messaging-topology-operator:1.8.0
- RabbitMQ: [e.g. 3.8.0]
- RabbitMQ Cluster Operator: rabbitmqoperator/cluster-operator:2.0.0
- Kubernetes: Kubernetes Version: v1.22.10
- Cloud provider or hardware configuration: On prem hardware + rancher + fleet + vmware
 
## Additional context
I did try this with an earlier version of both the MTO and the Cluster Operator, but tried updating them to the current versions to see if it resolved the issue. I don't have the specific older versions that it was also failing on documented.

**Comment by thomasletsch** [comment]
You need to specify the name of the container to overwrite:

```
spec:
  template:
    spec:
      containers:
        - command:
            - /manager
          name: manager
          env:
            - name: MESSAGING_DOMAIN_NAME
              value: "my.domain.com"
```

Added: **name: manager**

**Comment by thomasletsch** [comment]
What I am still struggling is the next step. Now the operator tries to connect to the adjusted domain name, but this DNS entry does not exist. 
As far as I understood it, this DNS should resolve to the internal IP of the node on which the rabbit cluster runs. How can we achieve this?

Message:               Put "https://rabbit-test.rabbitmq-system.svc.my.domain.com:15671/api/users/rabbit-user": dial tcp: lookup rabbit-test.rabbitmq-system.svc.my.domain.com on 10.43.0.10:53: no such host

And from the outside (loadbalancer or nodeport), the port 15671 is not accessible, since per default these ports are mapped to the 30.000 port range by k8n.

Since we are totally in an internal network, I would also be fine to call this without SSL, but I'm not a security expert :-)

**Comment by DanielePalaia** [comment]
Hi @thomasletsch sorry for the delay I just saw your message.

Indeed this environment variable MESSAGING_DOMAIN_NAME works when we are using a non default K8s domain name as we specified here:
https://rabbitmq.com/kubernetes/operator/install-topology-operator.html#non-default-k8s-domain

Indeed by default K8s creates domain name for services with that notation (servicename.namespace.svc).

Not sure at this point if for you is convenient to use a new certificate for the default dns, or to try to modify the domain name in k8s.

Currently we have this possibility described here https://www.rabbitmq.com/kubernetes/operator/using-topology-operator.html#non-operator, though seem not working correctly for TLS, as described here: https://github.com/rabbitmq/cluster-operator/issues/1185

We are planning to solve that issue but I'm not sure yet about a specific date.

Anyway as the original post of the Issue seems addressed I there are no objections I think we can close this related one.
